src/Condition.js

```diff
@@ -0,0 +1,233 @@
+    op: ' !',
+  },
+};
+
+class BooleanCondition {
+  constructor(condition, children) {
+    this._condition = condition;
+
+    switch (condition.type) {
+      case Affix.PREFIX:
+        this._children = transform(children);
+        break;
+      default:
+        this._children = children.map(child => transform(child));
+        break;
+    }
+  }
+
+  toVCL() {
+    let clause;
+
+    switch (this._condition.type) {
+      case Affix.PREFIX:
+        clause = this._children.toVCL();
+        return `${this._condition.op}(${clause})`;
+      default:
+        clause = this._children.map(child => child.toVCL()).join(this._condition.op);
+        return `(${clause})`;
+    }
+  }
+
+  evaluate(req) {
+    switch (this._condition.name) {
+      case 'and':
+        return this._children.reduce((result, child) => result && child.evaluate(req), true);
+      case 'or':
+        return this._children.reduce((result, child) => result || child.evaluate(req), false);
+      default:
+        return !this._children.evaluate(req);
+    }
+  }
+}
+
+/**
+ * PropertyCondition
+ */
+class PropertyCondition {
+  constructor(name, type, op, value, vcl, express) {
+    this._name = name;
+    this._type = type;
+    this._op = op;
+    this._value = value;
+    this._vcl = vcl;
+    this._express = express;
+  }
+
+  toVCL() {
+    const quote = this._type === 'string' ? '"' : '';
+    let value = this._value;
+    let op = this._op;
+
+    if (quote === '"' && !op) {
+      // substring-start
+      value = `^${value}`;
+      op = '~';
+    }
+    if (!op || op === '=') {
+      // operand defaults to equal
+      op = '==';
+    }
+    const vcl = typeof this._vcl === 'function' ? this._vcl(this) : this._vcl;
+    return `${vcl} ${op} ${quote}${value}${quote}`;
+  }
+
+  evaluate(req) {
+    const actual = this._express(req, this);
+    const value = this._value;
+
+    if (!actual) {
+      return false;
+    }
+
+    switch (this._op) {
+      case '=':
+        return actual === value;
+      case '~':
+        return !!actual.match(value);
+      case '<':
+        return actual < value;
+      case '>':
+        return actual > value;
+      default:
+        return actual.startsWith(value);
+    }
+  }
+
+  get name() {
+    return this._name;
+  }
+
+  get type() {
+    return this._type;
+  }
+}
+
+/**
+ * Known properties
+ */
+const propertyConditionMap = {
+  url: {
+    vcl: 'req.url',
+    express: req => req.path,
+    type: 'string',
+    allowed_ops: '=~',
+  },
+  url_param: {
+    vcl: (property) => {
+      let vcl = `subfield(req.url.qs, "${property.name}", "&")`;
+      if (property.type === 'number') {
+        vcl = `std.atoi(${vcl})`;
+      }
+      return vcl;
+    },
+    express: (req, property) => {
+      const { query } = url.parse(req.path, true);
+      return query[property.name];
+    },
+    allowed_ops: '~<=>',
+  },
+  client_lat: {
+    vcl: 'client.geo.latitude',
+    express: req => req.client_lat,
```

Comment:
where do you get this from?

Reply:
It's fake 😁 Mapping a client IP to those information requires an additional module, which I haven't selected yet

Reply:
For local development, we should treat these things as `true`, always. I don't think there are good services we can use that are free, fast, and work offline. And if there were, we would have to think how we can fake things.